nx-dev/nx-dev/pages/ai/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { ReactNode, useState } from 'react';
+import { DocumentationHeader } from '@nx/nx-dev/ui-common';
+import { useNavToggle } from '../../lib/navigation-toggle.effect';
+import { renderMarkdown } from '@nx/nx-dev/ui-markdoc';
+import {
+  getStringFromStream,
+  nxDevDataAccessAi,
+} from '@nx/nx-dev/data-access-ai';
+
+export default function AiDocs(): JSX.Element {
+  const { toggleNav, navIsOpen } = useNavToggle();
+
+  const [finalResult, setFinalResult] = useState('' as ReactNode);
+  const [error, setError] = useState(null);
+  const [query, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = async () => {
+    setLoading(true);
+    setFinalResult('');
+    try {
+      const response = await nxDevDataAccessAi(query);
+      const completeText = (await getStringFromStream(response)) as string;
+      setFinalResult(renderMarkdown(completeText, { filePath: '' })?.node);
```

Comment:
You always have a node:
```suggestion
      setFinalResult(renderMarkdown(completeText, { filePath: '' }).node;
```

Reply:
I would also extract the `setFinalResult` from your try and catch because the `completeText` needs to always be a `string` which is not the case here it is `string | Error`.